cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels.cl

```diff
@@ -408,6 +410,24 @@ DECLARE_SOURCE(
         }
     }
 
+    __kernel void count_empty_clusters(__global const int * partialCentroidsCounters, int K, int parts, __global int * numEmptyClusters) {
+        const int local_id   = get_local_id(1);
+        const int local_size = get_local_size(1);
+
+        int numEmpty = 0;
+        for (int i = local_id; i < K; i += local_size)
+        {
+            int count = 0;
+            for (int j = 0; j < parts; j++)
+            {
+                count += partialCentroidsCounters[j * K + i];
+            }
+            numEmpty += count > 0 ? 0 : 1;
+        }
```

Comment:
Inner loop iterates over the column. We can think about rearranging loops that inner will go on rows, probably it will give better performance

Reply:
I really need a result of inner cycle to be available in outer cycle